spec/models/article_spec.rb

```diff
@@ -463,11 +448,31 @@ def build_and_validate_article(*args)
       expect(unpublished_article.published_at).to be_nil
     end
 
-    it "does have a published_at if published" do
-      # this works because validation triggers the extraction of the date from the front matter
```

Comment:
Actually, the date is not extracted from the front matter in this specific case, but is set along with evaluating frontmatter. So the comment was a bit misleading, and I changed it.